exosphere-shared/src/docker-helper.ls

```diff
@@ -43,14 +44,19 @@ class DockerHelper
     child_process.exec-sync "docker rm -f #{container}" if @container-exists container
 
 
-  @run-image = (container, image) ->
-    if container is \test-mongo
-      child_process.exec-sync "docker run -d --name=#{container} -p 27017:27017 #{image}"
+  @run-image = (container, done) ~>
+    if container.container-name is \test-mongo
+      child_process.exec-sync "docker run -d --name=#{container.container-name} -p 27017:27017 #{container.image}"
     else
-      child_process.exec-sync "docker run -d --name=#{container} #{image}"
+      new ObservableProcess("docker run #{if container.volume? then that} --name=#{container.container-name} #{container.image}",
+                            stdout: {write: (text) -> },
+                            stderr: {write: (text) -> })
+        ..on 'ended', (exit-code) ->
+          | exit-code > 0  =>  throw new Error "Dependency #{container.container-name} failed to run"
```

Comment:
We should not throw exceptions. Call the `done` method with the error here, and return from the method. The caller must handle the error and exit.